### PR TITLE
Extend VM watch functest to verify printableStatus transitions

### PR DESCRIPTION
**Pull request by zcahana**
Signed-off-by: Zvi Cahana <zvic@il.ibm.com>

**What this PR does / why we need it**:

This PR complements #5528 by adding a functional test to verify the correct display of VM status via the CLI, across different VM lifecycle stages.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/vm_watch_test.go [hunk 1]
```diff
@@ -213,6 +214,77 @@ var _ = Describe("[rfe_id:3423][crit:high][vendor:cnv-qe@redhat.com][level:compo
...
+			"VM should be in the %s status", v12.VirtualMachineStatusRunning)
+
+		By("Migrating the VirtualMachine")
+
+		// Verify we have more than one scheduleable node
+		virtClient, err := kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+
+		nodes := tests.GetAllSchedulableNodes(virtClient)
+		if len(nodes.Items) > 1 {
```
Comment (enp0s3) [inline tests/vm_watch_test.go:261]: @zcahana I think that for the sake of coverage, we either should add another test case where we strictly demand more than 1 scheduable node in order to test the migration output, or we should demand it here in this test.

File: tests/vm_watch_test.go [hunk 2]
```diff
@@ -213,6 +214,77 @@ var _ = Describe("[rfe_id:3423][crit:high][vendor:cnv-qe@redhat.com][level:compo
 		Expect(err).ToNot(HaveOccurred())
 		Expect(titles).To(Equal([]string{"NAME", "AGE", "STATUS", "VOLUME"}),
 			"Output should have the proper columns")
+
+		vmStatus, err := readNewStatus(stdout, titles, readTimeout)
```
Comment (enp0s3) [inline tests/vm_watch_test.go:218]: @zcahana Could you PTAL at `readNewStatus` and see if it's possible to refactor it in a non recursive manner? if it's a big change then let's do it in a separate PR.

**Review by enp0s3 — CHANGES_REQUESTED** [review]
@zcahana Hi. Thank you! Please see my comments below.

File: tests/vm_watch_test.go (hunk 1 above)
Comment (zcahana) [reply tests/vm_watch_test.go:261]: Changed so that the current testcase requires #nodes >= 2.

File: tests/vm_watch_test.go (hunk 2 above)
Comment (zcahana) [reply tests/vm_watch_test.go:218]: Doesn't look like a big change. Changing...

**Comment by zcahana** [comment]
There gotta be some builtin detection in git from running silly things like `git commit -m --amend` :man_facepalming:

**Comment by zcahana** [comment]
@enp0s3 thanks for your review! Do you think there's anything else needed for this to be merged?

**Review by enp0s3 — COMMENTED** [review]
@zcahana Thank you for addressing the the comments! Another concern I have is that if something wrong will happen to the cluster during this test run, and state transition won't happen although the virtctl command will return success, we may stuck at the loop of readNewStatus forever

**Review by enp0s3 — APPROVED** [review]
/approve
